src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -1051,6 +1051,7 @@ class DocBaseViewer extends BaseViewer {
 
     loadUIReact() {
         this.controls = new ControlsRoot({ containerEl: this.containerEl });
+        this.annotationControlsFSM.subscribe(() => this.renderUI());
```

Comment:
Is the anonymous function needed? Would there be any benefit to combining the fsm callbacks?

Reply:
It's needed, otherwise I need to manually bind `this.renderUI` in the constructor. I think part of the value of the observer pattern is its support for distributed callbacks; combining them would be the equivalent of having a single container connected to a redux store for an entire app.